app/controllers/admin/lead_provider_users_controller.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+class Admin::LeadProviderUsersController < Admin::BaseController
+  before_action :set_lead_provider
+
+  def index
+    @users = @lead_provider.users
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(
+      first_name: params.dig(:user, :first_name),
+      last_name: params.dig(:user, :last_name),
+      email: params.dig(:user, :email),
+    )
+
+    ActiveRecord::Base.transaction do
+      @user.save!
+      LeadProviderProfile.create!(user: @user, lead_provider: @lead_provider)
+    end
+
+    redirect_to admin_lead_provider_users_path
+  rescue ActiveRecord::RecordInvalid
+    render :new
+  end
+
+  def edit
+    @user = User.find(params[:id])
```

Comment:
We can probably restrict this with pundit but even then scoping like @lead_provider.users.find would probably be better than just User.find